# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1226432042 18000
#      Tue Nov 11 14:34:02 2008 -0500
# Node ID 7124bffc89c706595d118183b2ac019c5116d241
# Parent  87f316e219b6aef227d06d0947ca811192dcbc51
parse.y (F__parser_debug_flag__): New function.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2008-11-11  John W. Eaton  <jwe@octave.org>
+
+	* parse.y (F__parser_debug_flag__): New function.
+
 2008-10-31  Jaroslav Hajek <highegg@gmail.com>
 
 	* xnorm.cc: New source.
 	* xnorm.h: New header file.
 	* Makefile.in: Include xnorm.cc in the build process.
 	* data.cc (Fnorm): Call xnorm, xcolnorms, xrownorms or xfrobnorm
 	to do the actual work.
 
diff --git a/src/parse.y b/src/parse.y
--- a/src/parse.y
+++ b/src/parse.y
@@ -4096,13 +4096,28 @@ context @var{context}, which may be eith
         error ("evalin: expecting string as first argument");
     }
   else
     print_usage ();
 
   return retval;
 }
 
+DEFUN (__parser_debug_flag__, args, nargout,
+  "Undocumented internal function.")
+{
+  octave_value retval;
+
+  bool debug_flag = octave_debug;
+
+  retval = set_internal_variable (debug_flag, args, nargout,
+           			  "__parser_debug_flag__");
+
+  octave_debug = debug_flag;
+
+  return retval;
+}
+
 /*
 ;;; Local Variables: ***
 ;;; mode: text ***
 ;;; End: ***
 */
